src/hotspot/share/services/nmt/memoryLogRecorder.cpp

```diff
@@ -0,0 +1,961 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "jvm.h"
+#include "nmt/nmtCommon.hpp"
+#include "nmt/mallocHeader.hpp"
+#include "nmt/memTracker.hpp"
+#include "runtime/globals.hpp"
+#include "runtime/os.hpp"
+#include "runtime/thread.hpp"
+#include "services/nmt/memoryLogRecorder.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/nativeCallStack.hpp"
+
+#if defined(LINUX)
+#include <malloc.h>
+#elif defined(__APPLE__)
+#include <malloc/malloc.h>
+#endif
+
+#ifdef ASSERT
+
+/*
```

Comment:
Style: Use `//` or:

```
/*
 * This
 * Pattern
 */
```

FWIW, I prefer `//`. I'm sure your editor has support for doing this.